jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletHandler.java

```diff
@@ -1731,6 +1689,71 @@ void destroyListener(EventListener listener)
             _contextHandler.destroyListener(listener);
     }
 
+    /**
+     * A mapping of a servlet
+     */
+    public static class MappedServlet
+    {
+        private final PathSpec _pathSpec;
+        private final ServletHolder _servletHolder;
+        private final ServletPathMapping _servletPathMapping;
+
+        MappedServlet(PathSpec pathSpec, ServletHolder servletHolder)
+        {
+            _pathSpec = pathSpec;
+            _servletHolder = servletHolder;
+
+            // Create the HttpServletMapping only once if possible.
+            if (pathSpec instanceof ServletPathSpec)
+            {
+                switch (pathSpec.getGroup())
+                {
+                    case EXACT:
+                    case ROOT:
+                        _servletPathMapping = new ServletPathMapping(_pathSpec, _servletHolder.getName(), _pathSpec.getPrefix());
+                        break;
+                    default:
+                        _servletPathMapping = null;
+                        break;
+                }
+            }
+            else if (_pathSpec == null)
+            {
+                // Named servlet mapping
```

Comment:
See my comment on the ServletPathMapping class: this comment at like 1722 is the only clue that the ServletPathMapping class can be used as a name-match only .

Reply:
fixed. see above.